src/app/telemetry/telemetry.ts

```diff
@@ -24,28 +26,30 @@ class Telemetry implements ITelemetry {
   public initialize() {
     if (this.instrumentationKey) {
       this.appInsights.loadAppInsights();
+      this.appInsights.addTelemetryInitializer(this.filterFunction);
       this.appInsights.trackPageView();
     }
   }
 
   public trackEvent(eventName: string, payload: any) {
-    if (!this.validateEventName(eventName)) {
-      throw new Error('Invalid telemetry event name');
-    }
-
     this.appInsights.trackEvent({ name: eventName }, payload);
   }
 
   public trackException(error: Error, severityLevel: SeverityLevel) {
     this.appInsights.trackException({ error, severityLevel });
   }
 
-  // A valid event name ends with the word EVENT
-  private validateEventName(eventName: string): boolean {
-    const listOfWords = eventName.split('_');
-    const lastIndex = listOfWords.length - 1;
-    const lastWord = listOfWords[lastIndex];
-    return lastWord === 'EVENT';
+  private filterFunction(envelope: any): boolean {
```

Comment:
Please add comment so the future knows why we do this and why it is important to maintain.

Reply:
The comment should state the intended use of this callback - we may want to use this in the future to filter which telemetry gets captured.